pkg/backend/display/display.go

```diff
@@ -37,10 +37,19 @@ func ShowEvents(
 		// TODO[pulumi/pulumi#2390]: enable JSON display for real deployments.
 		contract.Assertf(isPreview, "JSON display only available in preview mode")
 		ShowJSONEvents(op, action, events, done, opts)
-	} else if opts.DiffDisplay {
+		return
+	}
+
+	switch opts.Type {
+	case DisplayDiff:
 		ShowDiffEvents(op, action, events, done, opts)
-	} else {
+	case DisplayProgress:
 		ShowProgressEvents(op, action, stack, proj, events, done, opts, isPreview)
+	case DisplayQuery:
```

Comment:
Given that the only place we seem to consume `DisplayQuery` is this location where we Fail - is there any reason it needs to be part of the enum here - vs. some other flag orthogonal to this enum?  It feels like this isn't really another display mode in the same sense as the other two - though I may be misreading some of the implications of this.

Reply:
I'm not sure I understand the objection, but to me this does seem like it should be alongside diff and progressing view. It's basically a completely separate modality for the display settings, and I actually think that's really important to capture.